torch/jit/_script.py

```diff
@@ -1079,5 +1080,100 @@ def _unwrap_optional(x):
     return x
 
 
+def get_origin(target_type):
+    return getattr(target_type, "__origin__", None)
+
+
+def get_args(target_type):
+    return getattr(target_type, "__args__", None)
+
+
+def check_args_exist(target_type):
+    if target_type is List or target_type is list:
+        _jit_internal.is_list(target_type)
+    elif target_type is Tuple or target_type is tuple:
+        _jit_internal.is_tuple(target_type)
+    elif target_type is Dict or target_type is dict:
+        _jit_internal.is_dict(target_type)
+    elif target_type is None or target_type is Optional:
+        _jit_internal.is_optional(target_type)
```

Comment:
Here I'm using the _jit_internal.py functions only for the error message they throw, the return value of these function calls aren't used. I'm still questioning if I should use the _jit_internal.py functions this way or if I should extract the error messages from the _jit_internal.py functions, put them in a different location, and then have this part of the code for _isinstance and the _jit_internal.py functions call them separately

Reply:
I think either way is fine with a slight preference to refactor the error message part and reuse. Just to make sure I understand why we don't use those return values, it is because you are rolling your own logic to check `origin` and `typing` like on line 1108 and 1109?

Reply:
The _jit_internal.py functions don't aid in helping to check for nested containers, they only check if the outer container is of the right type and if the outer container contains an inner type. In generics_checker on 1102-1157 there's a need to do work to process nested containers and I think the code is cleaner when I don't use the _jit_intneral.py functions.

I'm also confused on what the checks for the `__module__` below code int he _jit_internal.py functions are doing:
```
if not hasattr(ann, '__module__'):
        return False
    return ann.__module__ == 'typing' and \
        (getattr(ann, '__origin__', None) is List or
            getattr(ann, '__origin__', None) is list)
```
Should I be checking for this `__module__` attribute in my logic somewhere? I'm not familiar enough yet with modules to know if I'm missing something here

Reply:
I don't think this is necessary since you are already checking equality against `List`, which is from `typing` module. I am not entirely sure why it is needed in _jit_internal.py. @SplitInfinity Could you clarify?

Reply:
Yeah, I'm not sure either.

I think it might also make sense to move this code to `jit_internal.py` to reuse those error messages. I would prefer that to calling `is_list`, etc. for the error message and throwing away the return value. `ignore`, `export`, `is_scripting` and `ignore` live there and those are certainly not internal APIs (but I think they are in that file in order to avoid circular import problems).

Reply:
Okay, I'll move the code and factor out the error messages 💯